File: MB-Secretaria/review/review-blocking.ts
import { REVIEW_REASONS } from "./review-reasons";
import type { BlockingResult, ClinicalReviewCase, HumanReviewFlags } from "./types";

export function evaluateReviewBlocking(reviewCase: ClinicalReviewCase, flags: HumanReviewFlags): BlockingResult {
  const reason: string[] = [];
  const highRisk = reviewCase.risk_level === "critical";
  const unsafeReason =
    highRisk ||
    reviewCase.review_reason.includes(REVIEW_REASONS.HALLUCINATION_RISK) ||
    reviewCase.review_reason.includes(REVIEW_REASONS.POLICY_BLOCK);

  if (unsafeReason) reason.push(REVIEW_REASONS.POLICY_BLOCK);

  const enforcementActive = flags.enabled && flags.blockingEnabled;

  return {
    blocked: enforcementActive && unsafeReason,
    unsafe_to_display: enforcementActive && unsafeReason,
    requires_override: unsafeReason,
    shadow_block_recommended: !enforcementActive && unsafeReason,
    reason,
  };
}

File: MB-Chat/review_py/review_blocking.py
from __future__ import annotations

from dataclasses import dataclass

from . import review_reasons
from .types import ClinicalReviewCase, HumanReviewFlags


@dataclass(slots=True)
class BlockingResult:
    blocked: bool
    unsafe_to_display: bool
    requires_override: bool
    shadow_block_recommended: bool
    reason: list[str]


def evaluate_review_blocking(review_case: ClinicalReviewCase, flags: HumanReviewFlags) -> BlockingResult:
    unsafe_reason = (
        review_case.risk_level == "critical"
        or review_reasons.HALLUCINATION_RISK in review_case.review_reason
        or review_reasons.POLICY_BLOCK in review_case.review_reason
    )
    reasons = [review_reasons.POLICY_BLOCK] if unsafe_reason else []
    enforcement_active = flags.enabled and flags.blocking_enabled
    return BlockingResult(
        blocked=bool(enforcement_active and unsafe_reason),
        unsafe_to_display=bool(enforcement_active and unsafe_reason),
        requires_override=bool(unsafe_reason),
        shadow_block_recommended=bool(not enforcement_active and unsafe_reason),
        reason=reasons,
    )
